src/ppx_deriving.mli

```diff
@@ -18,15 +18,34 @@ open Parsetree
     annotations. If this function is missing, the corresponding [[%foo:]]
     annotation is ignored. *)
 type deriver = {
+  name : string ;
   core_type : (core_type -> expression) option;
-  structure : options:(string * expression) list -> path:string list ->
-              type_declaration list -> structure;
-  signature : options:(string * expression) list -> path:string list ->
-              type_declaration list -> signature;
+  type_decl_str : options:(string * expression) list -> path:string list ->
+                   type_declaration list -> structure;
+  type_ext_str : options:(string * expression) list -> path:string list ->
+                   type_extension -> structure;
+  type_decl_sig : options:(string * expression) list -> path:string list ->
+                   type_declaration list -> signature;
+  type_ext_sig : options:(string * expression) list -> path:string list ->
+                  type_extension -> signature;
 }
 
-(** [register name deriver] registers [deriver] as [name]. *)
-val register : string -> deriver -> unit
+(** [register deriver] registers [deriver] according to its [name] field. *)
+val register : deriver -> unit
+
+(** Creating {!deriver} structure. *)
+val create :
+  string ->
+  ?core_type: (core_type -> expression) ->
+  ?type_ext_str: (options:(string * expression) list -> path:string list ->
+                   type_extension -> structure) ->
+  ?type_ext_sig: (options:(string * expression) list -> path:string list ->
+                   type_extension -> signature) ->
+  type_decl_str: (options:(string * expression) list -> path:string list ->
```

Comment:
I wonder what's the use of the rewriter if it does not handle type declarations. That's no problem for me, but making handling of type declarations mandatory makes it impossible to forget such a parameter, which can sometimes be a stupid error hard to think about.